api/arceos_posix_api/src/imp/stdio.rs

```diff
@@ -0,0 +1,155 @@
+use core::cell::UnsafeCell;
+
+use axerrno::AxResult;
+use axio::{prelude::*, BufReader, Result};
+
+#[cfg(feature = "fd")]
+use {alloc::sync::Arc, axerrno::LinuxResult, axio::PollState};
+
+fn console_read_bytes() -> Option<u8> {
+    axhal::console::getchar().map(|c| if c == b'\r' { b'\n' } else { c })
+}
+
+fn console_write_bytes(buf: &[u8]) -> AxResult<usize> {
+    axhal::console::write_bytes(buf);
+    Ok(buf.len())
+}
+
+struct StdinRaw;
+
+impl Read for StdinRaw {
+    fn read(&mut self, buf: &mut [u8]) -> AxResult<usize> {
+        let mut read_len = 0;
+        while read_len < buf.len() {
+            if let Some(c) = console_read_bytes() {
+                buf[read_len] = c;
+                read_len += 1;
+            } else {
+                break;
+            }
+        }
+        Ok(read_len)
+    }
+}
+
+pub struct Stdin {
+    inner: UnsafeCell<BufReader<StdinRaw>>,
```

Comment:
I revert `arceos-posix-api/stdio.rs`,  and it seems that `axsync` in `arceos-posix-api` should not be optional in `arceos-posix-api/Cargo.toml`, because `Stdin` and `Stdout` need it.